worlds/oot_soh/location_access/dungeons/deku_tree.py

```diff
@@ -37,113 +34,228 @@ def set_region_rules(world: "SohWorld") -> None:
 
     double_link_regions(world, Regions.DEKU_TREE_ENTRYWAY.value, Regions.DEKU_TREE_LOBBY.value)
 
-    # Slingshot Room access requires shield
-    world.get_region(Regions.DEKU_TREE_LOBBY.value).connect(
-        world.get_region(Regions.DEKU_TREE_SLINGSHOT_ROOM.value),
-        rule=lambda state: can_reflect_nuts(state, world) or can_use(Items.MEGATON_HAMMER.value, state, world))
-
-    world.get_region(Regions.DEKU_TREE_SLINGSHOT_ROOM.value).connect(
-        world.get_region(Regions.DEKU_TREE_LOBBY.value))
-
-    # Compass Room - simple access from lobby
-    double_link_regions(world, Regions.DEKU_TREE_LOBBY.value, Regions.DEKU_TREE_COMPASS_ROOM.value)
-
-    # Basement access requires ability to deal with enemies or adult
-    world.get_region(Regions.DEKU_TREE_LOBBY.value).connect(
-        world.get_region(Regions.DEKU_TREE_BASEMENT_LOWER.value),
-        rule=lambda state: is_adult(state, world) or can_attack(state, world) or can_use(Items.NUTS.value, state, world))
-
-    world.get_region(Regions.DEKU_TREE_BASEMENT_LOWER.value).connect(
-        world.get_region(Regions.DEKU_TREE_LOBBY.value))
-
-    world.get_region(Regions.DEKU_TREE_BASEMENT_LOWER.value).connect(
-        world.get_region(Regions.DEKU_TREE_BASEMENT_BACK_ROOM.value),
-        # rule=lambda state: has_fire_source_with_torch(state, world) or can_use(Items.FAIRY_BOW.value, state, world))
+    # repeated rules
+    normal_grass_rule = lambda state: can_cut_shrubs(state, world)
+    requires_reflect_grass_rule = lambda state: can_cut_shrubs(state, world) and can_reflect_nuts(state, world)
+    #requires_fire_grass_rule = lambda state: can_cut_shrubs(state, world) and has_fire_source_with_torch(state, world)
+            # Above commented out because it can't succeed without stick pot event implemented
+    requires_fire_grass_rule =lambda state: can_cut_shrubs(state, world)
```

Comment:
We shouldn't introduce helpers that SoH doesn't also have to keep logic as 1 to 1 as possible. Not to mention these feel very redundant as well.